### DISCUSSION TITLE
Not able to send webhook alert , always getting 400 bad request on Grafana v9.3.4 while same request is working on postman

### DISCUSSION THREAD

**Discussion by sandeepsonitwi** [grafana/grafana#62684] [Alerting]
Hi Team,

I am using Grafana v9.3.4 via docker, I am trying to send webhook alert to a URL , when i am clicking on "TEST" button on contact-point creation window it always gives me 400 bad request. Previously i was doing same thing in Grafana 9.3.2 version where it was not sending payload in post request. So please let me know how can i debug this issue or how can check what is the exact error. 

<img width="1718" alt="Screenshot 2023-02-01 at 6 22 15 PM" src="https://user-images.githubusercontent.com/122768877/216048196-28d1e1aa-c15a-406b-90f6-48c8824dde04.png">

Below is same request which is working fin via postman
<img width="1177" alt="Screenshot 2023-02-01 at 6 26 06 PM" src="https://user-images.githubusercontent.com/122768877/216049818-7723f8d7-bb29-4ac4-98c6-524d3f11f5f5.png">

Below is network call details from Grafana where it is displaying all request data properly but still response status is 400 
<img width="1668" alt="Screenshot 2023-02-01 at 6 32 02 PM" src="https://user-images.githubusercontent.com/122768877/216050801-7cafa56b-2126-4ece-b3c8-975c89e41ad2.png">
<img width="1519" alt="Screenshot 2023-02-01 at 6 44 14 PM" src="https://user-images.githubusercontent.com/122768877/216053177-7e6ffa9c-18f5-46c1-8e8b-9fd55c7a3d78.png">

Please let me know how can i debug this further, when these alerts are fired by rule section then how can i check logs to see what is the response of webhook.

**Comment by somratdnutanix** [comment]
i am facing the same problem, any updates on this one?

**Reply by sandeepsonitwi** [reply]
@somratdnutanix  you can try above answer, I am able to get alert data using this.

**Reply by somratdnutanix** [reply]
thank you so much, issue is resolved

**Reply by GNK-Cloud** [reply]
How did you slove the issue?. I am also facing same issue. Pls share your template or steps

**Comment by sandeepsonitwi** [comment]
I found that Grafana always sends a predefined payload to configured URL. So we cannot modify this payload, while if we need we can add additional payload in message field. Below is the sample json which we will receive in webhook call, So we need to create a object for below json in POST call to receive alerts. Please note this works only in 9.3.6 version.

{"receiver": "webhook-local-contact-point",
"status": "firing",
"orgId": 1,
"alerts": [
{
"status": "firing",
"labels": {
"alertname": "web-app-down",
"team": null,
"zone": null
},
"annotations": {
"description": null,
"runbook_url": null,
"summary": null
},
"startsAt": 1675330740000,
"endsAt": -62135769600000,
"generatorURL": "http://localhost:3000/alerting/grafana/O7uQpK0Vk/view",
"fingerprint": "defaa9e02d1b4104",
"silenceURL": "http://localhost:3000/alerting/silence/new?alertmanager=grafana&matcher=alertname%3Dweb-app-down&matcher=grafana_folder%3Dapp-alerts&matcher=instance%3D127.0.0.1%3A9191&matcher=job%3Dspring-actuator&matcher=web-app-status%3Ddown",
"dashboardURL": "",
"panelURL": "",
"valueString": "[ var='B' labels={name=up, instance=127.0.0.1:9191, job=spring-actuator} value=0 ], [ var='C' labels={name=up, instance=127.0.0.1:9191, job=spring-actuator} value=1 ]"
}
],
"commonLabels": {
"team": null
},
"externalURL": "http://localhost:3000/",
"version": "1",
"groupKey": "{}:{alertname="web-app-down", grafana_folder="app-alerts"}",
"truncatedAlerts": 0,
"title": "[FIRING:1] web-app-down app-alerts (127.0.0.1:9191 spring-actuator down)",
"state": "alerting",
"message": ""
}

**Comment by gillesdemey** [comment]
Hi! You can find an annotated payload example in our documentation 👉 https://grafana.com/docs/grafana/latest/alerting/manage-notifications/webhook-notifier/

**Reply by somratdnutanix** [reply]
actually the webhook payload format for 9.3.6 is different that this, it has an additional field called `values`.
This documentation might be old, it would be kind if you can update it.

**Comment by Sam-H101** [comment]
one way I have gotten around this is to use a "middleman" service such as flask. 

I will send the webhook message to the flask endpoint then using python, parse it out and repackage it to how I want it to be sent and send it on its way.

**Comment by gelicia** [comment]
This looks like it was resolved. Closing this discussion.
